packages/block-editor/src/components/inner-blocks/use-inner-block-template-sync.js

```diff
@@ -112,5 +110,5 @@ export default function useInnerBlockTemplateSync(
 		return () => {
 			isCancelled = true;
 		};
-	}, [ innerBlocks, template, templateLock, clientId ] );
+	}, [ template, templateLock, clientId ] );
```

Comment:
Note the condition on this line:

https://github.com/WordPress/gutenberg/pull/58348/files#diff-dedab2aa78fc2066d4eb8492c63d4cd6ea9ce9db9a2d2a5994755eabc4f02b4dR82

Nothing happens unless the template changed.